lib/BackgroundJob/RemoveEmptyRooms.php

```diff
@@ -77,7 +81,7 @@ public function callback(Room $room): void {
 			return;
 		}
 
-		if ($this->participantService->getNumberOfActors($room) === 0 && $room->getObjectType() !== 'file') {
+		if ($this->participantService->getNumberOfActors($room) === 0 && $room->getObjectType() !== 'file' && $this->federationManager->getNumberOfInvitations($room) === 0) {
```

Comment:
When there is no local user in the room anymore it should be delete. Or is this for the share-receiving side?

Reply:
This is for the share-receiving side. When a share for an unknown room is received, it is created on the rooms table then the invitation put in the invitations table so don't delete rooms that have pending invitations for it.